lib/slave/src/ESM.cc

```diff
@@ -0,0 +1,96 @@
+#include "kickcat/ESM.h"
+#include <algorithm>
+#include "protocol.h"
+
+namespace kickcat::ESM
+{
+    AbstractState::AbstractState(uint8_t id, AbstractESC& esc, PDO& pdo)
+        : id_(id)
+        , esc_{esc}
+        , pdo_{pdo}
+    {
+    }
+
+    void AbstractState::set_mailbox(mailbox::response::Mailbox* mbx)
+    {
+        mbx_ = mbx;
+    }
+
+    uint8_t AbstractState::id()
+    {
+        return id_;
+    }
+    void AbstractState::on_entry(Context, Context){};
```

Comment:
I think that default stub implementation are better placed in the header to emphasize the fact that it is really optional.